### PR TITLE
Contribution confirmation page should not display the name of payment processor type

### PR DESCRIPTION
**Pull request by agileware-pengyi**
Overview
----------------------------------------
The name of the payment processor type is displayed on the confirmation page. That is not needed.

Before
----------------------------------------
The text is **Click the Continue button to go to PayPal_Standard, where you will select your payment method and complete the contribution.**
confirm-page

After
----------------------------------------
The text is **Click the Continue button to complete the contribution.**

Comments
----------------------------------------
A fixed text is good enough.
It can display the name/title of the payment processor instance, however, there is no description on the payment processor settings page says the field will be displayed publicly and the site admin has to update it.

Gitlab ref: https://lab.civicrm.org/dev/financial/-/issues/134

Agileware ref: CIVICRM-1496

### CODE REVIEW

**Comment by yashodha** [comment]
@agileware-pengyi 

Can you please incorporate Eileen's feedback so that we can merge this?

File: CRM/Core/Payment.php
Comment (agileware-justin) [inline CRM/Core/Payment.php:599]: Recommend removing this line and logic entirely.

File: CRM/Core/Payment.php
Comment (agileware-justin) [inline CRM/Core/Payment.php:601]: Remove this logic

File: CRM/Core/Payment.php
Comment (agileware-justin) [inline CRM/Core/Payment.php:602]: OK with changing text to: "button to proceed with the contribution" - minor text change

File: CRM/Core/Payment.php
Comment (agileware-justin) [inline CRM/Core/Payment.php:604]: This logic and return text is essentially the same as the others, so just remove it.

File: CRM/Core/Payment.php
Comment (agileware-justin) [inline CRM/Core/Payment.php:605]: And then below this we have a catch-all return text, which again is essentially the same. So just have a single catch-all return text, remove all other logic.

Makes things much simpler.

**Review by agileware-justin — CHANGES_REQUESTED** [review]

**Comment by agileware-justin** [comment]
> 1. add the getText function to the Paypalimpl file, overriding the parent function & move this the

@eileenmcnaughton but why do this? Don't we want to reduce the complexity and technical debt, not add more?

**Comment by agileware-justin** [comment]
@yashodha there's a lot of redundant logic and various for the same return message in this case statement, look beyond what this PR to change and you'll see quite a confusing mess.

**Comment by eileenmcnaughton** [comment]
@agileware-justin - so the wording exists because obviously people thought it was much clearer in the context of Paypal & it's in the Core_Payment class because paypal logic used to be pretty much everywhere.

We now have 2 possibilities
1) the wording belongs to paypal & makes sense there - in which case our model is the Paypal class overrides the parent & puts it it's specific wording 

OR
2) people generally agree that we can change the wording without it being less clear to paypal users - in which case we need some broader buy in on that as a User experience change

**Comment by agileware-justin** [comment]
@eileenmcnaughton did you see that the same user-facing text is repeated 4 times?

**Comment by eileenmcnaughton** [comment]
If we can add an after screenshot then we can ping some other people & check they are comfortable with the text change. Obviously the text was done like that because someone decided it was more logical for the user experience. I don't have strong opinion as to whether it is or not but need to see that others agree before changing as people often care about this sort of thing.

**Comment by agileware-justin** [comment]
> Can you please incorporate Eileen's feedback so that we can merge this?

So if we had done your changes then this would have been merged.
But now it needs approval and buy-in from others, OK @eileenmcnaughton - no worries.

I do genuinely want to know if have reviewed this section of the code and seen the redundant logic which is used to essentially repeat variations of the same message to the user.

It's like this section of code has been edited by multiple people without scrolling up or down beyond the lines of their work and may not have understood that the words used in their message text are essentially the same.

**Comment by eileenmcnaughton** [comment]
@agileware-justin so the changes I suggested - basically overriding the function in the paypal code would have meant the text didn't change - just the word paypal was fixed - in that case it just fixes a bug.

For an actual text change I wouldn't take it on myself to merge without ensuring others agree

**Comment by eileenmcnaughton** [comment]
(& re the code -I have s really focussed on the above - ie. determining if a text change is being made or just the wrong word is being fixed - I haven't done a code review on this at this stage)

**Comment by agileware-justin** [comment]
@eileenmcnaughton all good, love your work as always :)

**Comment by mlutfy** [comment]
Personally I'm all for simplifying, but would it be possible to clarify the PR description to fully reflect the changes before/after?

For example, there are comments about "remove this code if amount < 0". Why? What?

(I don't object, I just don't understand the code comments, and the before/after, and what was really changed.)

I personally don't like the confirmation pages, but I can understand confused users, if it says "click to complete your contribution", but they haven't entered any payment details yet. Can't we display the "Payment Processor label" (which was added a few releases ago?)

**Comment by MegaphoneJon** [comment]
@agileware-justin You don't have a screenshot handy of the repeated text, do you?  I don't think I have and PayPal Std clients at present.

**Comment by agileware-justin** [comment]
> Personally I'm all for simplifying, but would it be possible to clarify the PR description to fully reflect the changes before/after?

@mlutfy only the author can edit the PR text - github permissions

> For example, there are comments about "remove this code if amount < 0". Why? What?
> (I don't object, I just don't understand the code comments, and the before/after, and what was really changed.)

This section of code has redundant logic which is used to repeat variations of the same instructions to the user. Let me illustrate:
1. Repeat the same instructions
2. Same instructions are repeated
3. The instructions are the same
4. Makes no sense, why are the instructions the same?

To be honest, I would just display "Pay now" as a button and not have any text at all. But that's too controversial a change :)

> I personally don't like the confirmation pages, but I can understand confused users, if it says "click to complete your contribution", but they haven't entered any payment details yet. Can't we display the "Payment Processor label" (which was added a few releases ago?)

"Payment Processor label" is an optional field, so it cannot be relied upon to be set.

> You don't have a screenshot handy of the repeated text, do you? I don't think I have and PayPal Std clients at present.

@MegaphoneJon I'm not sure what screenshot you want to see. The repeated text we're trying to remove is visible here on the left hand side of the change, https://github.com/civicrm/civicrm-core/pull/17568/files

**Comment by eileenmcnaughton** [comment]
We discussed this today (@seamuslee001 @kcristiano @jusfreeman  + me) & a version of the text was agreed. @jusfreeman will update & @seamuslee001 will merge (based on him having paid more attention to the final agreement than me )

**Comment by agileware-justin** [comment]
@seamuslee001 ready to merge when you are mate!

**Pull request merged**
